### PR TITLE
Small polishing following PR #10124

### PR DESCRIPTION
**Pull request by AltGr**
* re-add the "native toplevel" message when running the native toplevel.
* simplify internal functor followinf @Drup's review

### CODE REVIEW

File: toplevel/native/topeval.ml [hunk 1]
```diff
@@ -32,6 +32,8 @@ let _dummy = (Ok (Obj.magic 0), Err "")
 external ndl_run_toplevel: string -> string -> res
   = "caml_natdynlink_run_toplevel"
 
+let implementation_label = " - native toplevel"
```
Comment (gasche) [inline toplevel/native/topeval.ml]: I'm sorry to be that nitpicking guy again, but I would use `Some "native toplevel"` and decide on the dash at render-time.

**Review by gasche — APPROVED** [review]
Approved in principle, see minor comment. Thanks!

**Comment by gasche** [comment]
(It looks like you need to promote some testsuite files who depend on the locations within the changed module.)

File: toplevel/topeval.mli [hunk 2]
```diff
@@ -32,6 +32,8 @@ open Format
 val getvalue : string -> Obj.t
 val setvalue : string -> Obj.t -> unit
 
+val implementation_label: string
```
Comment (ghost) [inline toplevel/topeval.mli:36]: I feel like this deserve a comment:

```suggestion
(** String displayed after [OCaml version XXX] when starting the toplevel. *)
val implementation_label: string
```

Comment (ghost) [reply toplevel/topeval.mli:36]: (or what @gasche said, just saw his comment)

**Pull request merged**